src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/CommonTestResourcesAccessTest.kt

```diff
@@ -14,18 +19,18 @@ class CommonTestResourcesAccessTest : FunSpec({
     }
 
     test("stringFromResources - file doesn't exist") {
-        shouldThrow<IllegalArgumentException> {
+        shouldThrow<IOException> {
             stringFromResources("/test/resource/does-not-exist.txt")
         }.also {
-            it.message shouldBe "File 'does-not-exist.txt' doesn't exist in directory '/resources/test/resource/'!"
+            it.message shouldBe "Cannot read test/resource/does-not-exist.txt"
         }
     }
 
     test("stringFromResources - directory doesn't exist") {
-        shouldThrow<IllegalArgumentException> {
+        shouldThrow<IOException> {
             stringFromResources("/this/resource/for/sure/does/not/exist.txt")
         }.also {
-            it.message shouldBe "Directory 'this' doesn't exist in directory '/resources/'!"
+            it.message shouldBe "Cannot read this/resource/for/sure/does/not/exist.txt"
         }
     }
```

Comment:
The error messages become a bit less helpful here, but I did it to make the implementation simpler. If we wanted to preserve such detailed messages, I feel the code would become complex, and it's not worth it IMO. These cases are still handled gracefully, which I think is the most important thing.